torch/_dynamo/output_graph.py

```diff
@@ -1003,11 +1011,12 @@ def __init__(self, output_graph, parent=None):
         # list to determine that, when tracing the body function of a HigherOrderOperator,
         # if a new proxy is actually a free variable.
         self.seen_proxies = set({})
-        # A list of previously free variables that we lifted to being inputs
-        # of the graph. If we are tracing a HigherOrderOperator's body_fn,
-        # then we need to keep track of this so we can rewrite the
-        # HigherOrderOperator call using the traced body_fn.
-        self.lifted_freevars = set({})
+        # A list of previously free variables that we lifted to being inputs of
+        # the graph. If we are tracing a HigherOrderOperator's body_fn, then we
+        # need to keep track of this so we can rewrite the HigherOrderOperator
+        # call using the traced body_fn. This is a list (instead of set) so that
+        # we can maintain the order of args for the HigherOrderOperator call.
+        self.lifted_freevars = list()
```

Comment:
Use dict and have keys w/ None values, that gives you both ordering and uniqueness.

Reply:
Done